### PR TITLE
Dynamic Dashboard: Tracking events M1

### LINKED ISSUE #12877: Dynamic Dashboard: Analytics
- [x] Tracking events for M2 cards #12891
- [x] Tracking events for M1 cards #12922
- [x] Fix duplicate tracking events in Blaze card #12963

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Closes: #12877 

## Description

Adds tracking events for Dynamic Dashboard M1 cards.

This PR adds the new tracking event `dynamic_dashboard_card_interacted ` that we introduced during M2 cards development to M1 cards for better consistency. 

## Testing instructions

- Create a JN site
- Create products and orders
- Login into the app using the JN store

Store onboarding
- Tap any of the tasks and confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `store_setup`. 
- Navigate back and tap "View all tasks" button.
- Confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `store_setup`. 

Performance
- Add "Performance" card to the dashboard
- Tap "View all store analytics" button 
- Confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `performance`. 
- Switch time range and confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `performance`. 
- Interact with the performance chart and confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `performance`. 

Top performers
- Add "Top performers" card to the dashboard
- Tap "View all store analytics" button 
- Confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `top_performers`. 
- Switch time range and confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `top_performers`. 
- Tap on a top-performing product and confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `top_performers`. 

Blaze campaigns
- Add "Blaze campaigns" card to the dashboard
- Tap "View all campaigns" button 
- Confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `blaze`. 
- Interact with the Blaze campaign card (tap on a product or campaign)
- Confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `blaze`. 

## Screenshots

| Store onboarding | Performance | Top performers | Blaze campaigns |
|--------|--------|--------|--------|
| image | image | image | image | 

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift [hunk 1]
```diff
@@ -146,10 +146,12 @@ final class BlazeCampaignDashboardViewModel: ObservableObject {
     }
 
     func didSelectCampaignList() {
+        analytics.track(event: .DynamicDashboard.dashboardCardInteracted(type: .blaze))
```
Comment (itsmeichigo) [inline WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift:149]: ❓ This method doesn't seem to be triggered from my testing. Is it redundant?

File: WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift [hunk 2]
```diff
@@ -160,6 +162,7 @@ final class BlazeCampaignDashboardViewModel: ObservableObject {
     }
 
     func didSelectCreateCampaign(source: BlazeSource) {
+        analytics.track(event: .DynamicDashboard.dashboardCardInteracted(type: .blaze))
```
Comment (itsmeichigo) [inline WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift:165]: ❓ Same question as above about the method not being triggered.

**Review by itsmeichigo — APPROVED** [review]
This works as expected for me 🚢  I left some questions about methods that don't seem to be triggered - we might need to revise the tracking done in those.

File: WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift (hunk 1 above)
Comment (selanthiraiyan) [reply WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift:149]: You can see these methods being triggered in the attached video. 

https://github.com/woocommerce/woocommerce-ios/assets/524475/baf9e604-a1b7-4e74-b71f-8c3245ac6d7f

---

This `didSelectCampaignList` method is triggered upon tapping the "View all campaigns" button.

File: WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift (hunk 2 above)
Comment (selanthiraiyan) [reply WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift:165]: This event is triggered upon tapping on the Blaze campaign in the dashboard card.

**Comment by selanthiraiyan** [comment]
Thanks for the review, Huong! Enabling auto-merge now. 

---

>  left some questions about methods that don't seem to be triggered - we might need to revise the tracking done in those.

I answered your questions by explaining when those two methods are triggered. Please let me know if I am missing something.

File: WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift:149]: Oh that's true! But I see that tapping on View all campaigns also triggers `configureBlazeSection` in `DashboardViewHostingController`, and both are tracking the same event. Should we remove one of these?

File: WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift (hunk 2 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift:165]: From my testing, tapping Create campaign triggers only `createBlazeCampaignTapped` in `DashboardViewHostingController`. I'm confused why we have two places to track the interaction for this button. Can you check again @selanthiraiyan?

**Comment by itsmeichigo** [comment]
Hey @selanthiraiyan - I responded in the threads above about tracking Blaze events. Please check them out when you can.

**Comment by selanthiraiyan** [comment]
Thanks, Huong. I created a PR to fix this. #12963

**Pull request merged**